Overhaul Dictionary Documentation

Mickeon opened the pull request:
Continuation of especially https://github.com/godotengine/godot/pull/69451 and https://github.com/godotengine/godot/pull/68838...

- Made the wording match how other classes are documented a **lot** more closely;
- Made use of _`[param]`_ and several other strong references more often;
- Changed words where it felt necessary, where they could be mistaken for another concept of Godot's API, where technical terms are more appropriate. For example:
	- "key/value pair" -> "entry".
		- I've most commonly seen people referring to a Dictionary's element as an "entry". Discussions are welcome, please.
- Modified examples extensively, with the main goal to improve readability and variety:
	- Removed quotes around results when they could be misunderstood as strings (`# Prints "5"` -> `# Prints 5`);
	- Improved consistency all around;
	- Added extra padding before comments.
	- Avoid overly long lines.
- **COMPLETELY** stripped away the part about Dictionary comparisons and hashes from the leading description.
	- Way to give nausea to users at first sight. Not that Dictionaries themselves are compared THAT much anyhow. Sheesh...

- Updated `Dictionary(from)` constructor's description to be correct.
	- This is temporary. It's not something desired, and should either be modified to work as intended (copy), or removed. See https://github.com/godotengine/godot/pull/69117.

-----------------------------
Feedback is very, very welcome.

Mickeon committed:
Overhaul Dictionary Documentation

Mickeon commented on doc/classes/Dictionary.xml:
```diff
@@ -200,123 +146,140 @@
+				if "Godot" in {"Godot": 4}:
+				    print("The key is here!") # Will be printed.
+				[/codeblock]
+				[b]Note:[/b] This method returns [code]true[/code] as long as the [param key] exists, even if its corresponding value is [code]null[/code].
```
This small note may or may not be mergeable with the very first sentence. Something along the lines of _"Returns [code]true[/code] if the dictionary contains an entry with the given [param key], regardless of its corresponding value"_

mhilbrunner reviewed (approved):
Changes look good, spotted nothing so far. Will let it stay open for a bit longer and merge if not one else notices anything.

Mickeon commented:
I should've left it as a draft as I also wanted to revamp the leading description. It's better to merge it ASAP though I suppose.

raulsntos commented on doc/classes/Dictionary.xml:
```diff
@@ -192,131 +138,148 @@
+				    {210, null},
 				}
+
+				GD.Print(myDict.Contains("Godot")) # Prints true
+				GD.Print(myDict.Contains(210))     # Prints true
+				GD.Print(myDict.Contains(4))       # Prints false
```
```suggestion
				GD.Print(myDict.Contains("Godot")); // Prints true
				GD.Print(myDict.Contains(210));     // Prints true
				GD.Print(myDict.Contains(4));       // Prints false
```

raulsntos commented on doc/classes/Dictionary.xml:
```diff
@@ -192,131 +138,148 @@
-				// You have to use Contains() here as an alternative to GDScript's `in` operator.
-				if (new Godot.Collections.Dictionary{{"godot", "engine"}}.Contains("godot"))
+				var myDict = new Godot.Collections.Dictionary
 				{
-				    // I am executed.
+				    {"Godot", 4},
+				    {210, null},
 				}
```
Missing semicolon (`;`) at the end. Since `Variant` is a struct, you can't use `null`, the equivalent would be a default-initialized Variant, I recommend using the `default` keyword for that.

Also I think the preferred style is to have spaces between the braces but we are probably not very consistent in this throughout the documentation.

```csharp
var myDict = new Godot.Collections.Dictionary
{
    { "Godot", 4 },
    { 210, default },
};
```

raulsntos commented on doc/classes/Dictionary.xml:
```diff
@@ -192,131 +138,148 @@
+				var dict2 = new Godot.Collections.Dictionary{{"A", 10}, {"B", 2}};
+
+				// Godot.Collections.Dictionary has no Hash() method. Use GD.Hash() instead.
+				GD.Print(GD.Hash(dict1) == GD.Hash(dict2)); # Prints true
```
```suggestion
				GD.Print(GD.Hash(dict1) == GD.Hash(dict2)); // Prints true
```

Mickeon commented:
Updated. Also added a small instruction on how to iterate a dictionary in the leading description, because if I didn't I would've later felt guilty about it.

Mickeon commented:
Updated. Also added a small instruction on how to iterate a dictionary in the leading description, because if I didn't I would've later felt guilty about it.

raulsntos commented on doc/classes/Dictionary.xml:
```diff
@@ -118,63 +119,14 @@
-		[b]Note:[/b] When declaring a dictionary with [code]const[/code], the dictionary itself can still be mutated by defining the values of individual keys. Using [code]const[/code] will only prevent assigning the constant with another value after it was initialized.
+		The entries of a dictionary can be iterated with the [code]for[/code] keyword:
+		[codeblock]
+		var groceries = {"Orange": 20, "Apple": 2, "Banana": 4}
+		for fruit in groceries:
+		    var amount = groceries[fruit]
```
You could also add the equivalent in C#:

```csharp
foreach (var (fruit, amount) in groceries)
{
    // ¯\_(ツ)_/¯
}
```

Alternatively:

```csharp
foreach (var entry in groceries)
{
    var amount = entry.Value;
}
``` 

But I think we should prefer the deconstructing syntax (first option) in documentation examples.

raulsntos commented on doc/classes/Dictionary.xml:
```diff
@@ -118,63 +119,14 @@
-		[/csharp]
-		[/codeblocks]
-		[b]Note:[/b] When declaring a dictionary with [code]const[/code], the dictionary itself can still be mutated by defining the values of individual keys. Using [code]const[/code] will only prevent assigning the constant with another value after it was initialized.
+		The entries of a dictionary can be iterated with the [code]for[/code] keyword:
```
From looking at the example it looks like GDScript iterates the keys of the dictionary, not the entries.

Mickeon commented:
Thank you very much for the feedback

Mickeon commented on doc/classes/Dictionary.xml:
```diff
@@ -118,63 +119,23 @@
-		public Godot.Collections.Array array2 = new Godot.Collections.Array{1, 2, 3};
-
-		public void CompareArrays()
+		var groceries = {"Orange": 20, "Apple": 2, "Banana": 4}
```
I should've noticed myself, thank you very much

akien-mga commented:
Thanks!
